test/envtest/setup.go

```diff
@@ -0,0 +1,85 @@
+package envtest
+
+import (
+	"go/build"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"sync"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+// TODO: extract the version from go.mod:
+// https://github.com/Kong/gateway-operator/issues/548
+const kongConfVersion = "v0.0.10"
+
+// Setup sets up a test k8s API server environment and returned the configuration.
+func Setup(t *testing.T, scheme *k8sruntime.Scheme) *rest.Config {
```

Comment:
Might make a note that this was inspired by https://github.com/Kong/kubernetes-ingress-controller/blob/82ed3db914b79b8f04d2ddf5fc097373db4d39c8/test/envtest/setup.go#L65-L125

Reply:
Added comments.